Swiften/Parser/PayloadParsers/ClientStateParser.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2017 Tarun Gupta.
+ * Licensed under the simplified BSD license.
+ * See Documentation/Licenses/BSD-simplified.txt for more information.
+ */
+
+#include <Swiften/Parser/PayloadParsers/ClientStateParser.h>
+
+namespace Swift {
+
+ClientStateParser::ClientStateParser() : level_(0) {
```

Comment:
You could also initialise **level_** in the header by simply assigning 0 to the member there. This makes it easier to spot uninitialised members.